doc/src/suite-name-reg.rst

```diff
@@ -22,6 +25,17 @@ which creates the suite run directory structure and some service files
 underneath it. Otherwise, ``cylc run`` will do this at suite start up.
 
 
+.. _SuiteNames:
+
+Suite Names
+-----------
+
+Suite names are not validated. Names for suites can be anything that is a
+`valid filename <https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations>`_ within your operating system's file system,
+e.g. for Linux any name that does not contain any forward slash characters
+(``/``) and is not too long (as described under :ref:`TaskNames`).
```

Comment:
Suite names *can* contain `/` for hierarchical suite names. We should recommend users to avoid space characters, etc.

Reply:
> Suite names can contain / for hierarchical suite names.

I know they can, but that aspect is not documented except as part of the ``cylc register`` command ``--help`` i.e. docstring, so to mention it would have to write a whole new paragraph at least explaining hierarchical suites. I thought we want this in ASAP, so I don't want to keep changing the scope of this PR as I want to finalise it & get it merged before next release?

> We should recommend users to avoid space characters, etc.

Okay, I'll get this incorporated.

Reply:
I've done the best I can for nowwith regards to ``/`` in suite names, by cross-referencing the ``cylc register`` help content.